pennylane/qchem/openfermion_obs.py

```diff
@@ -917,6 +892,124 @@ def molecular_hamiltonian(
     + (0.176276408043196) [Z2 Z3]
     """
 
+    if len(args) != 0:
+        return _molecular_hamiltonian_dispatch(*args, **kwargs)
+
+    if "symbols" in kwargs:
+        symbols = kwargs["symbols"]
+        kwargs.pop("symbols")
+        return _molecular_hamiltonian_dispatch(symbols, **kwargs)
+
+    if "molecule" in kwargs:
+        molecule = kwargs["molecule"]
+        kwargs.pop("molecule")
+        return _molecular_hamiltonian_dispatch(molecule, **kwargs)
```

Comment:
I think you can refactor this into a single if statement.
`key = "symbols" if "symbols" in kwargs else "molecule" if "molecule" in kwargs else None`
`if key:` 
    `   value = kwargs.pop(key)`
    `   return _molecular_hamiltonian_dispatch(value, **kwargs)`

Reply:
Awkward, @obliviateandsurrender and I gave roughly the same idea.

Reply:
Great minds think alike. :)